include/deal.II/base/tensor.h

```diff
@@ -1016,6 +1043,20 @@ constexpr inline DEAL_II_ALWAYS_INLINE
 }
 
 
+
+template <int dim, typename Number>
+inline std::array<Number, Tensor<0, dim, Number>::n_independent_components>
+Tensor<0, dim, Number>::to_array() const
+{
+  std::array<Number, Tensor<0, dim, Number>::n_independent_components> result;
+  Assert(dim != 0,
+         ExcMessage("Cannot access an object of type Tensor<0,0,Number>"));
+  result[0] = value;
+  return result;
```

Comment:
I think you can just do `return {value};`, or possibly with another set of braces.